pkg/router/controller/contention.go

```diff
@@ -258,15 +256,32 @@ func ingressEqual(a, b *routev1.RouteIngress) bool {
 }
 
 // ingressConditionsEqual determines if the route ingress conditions are equal,
-// while ignoring LastTransitionTime
+// while ignoring LastTransitionTime.
 func ingressConditionsEqual(a, b []routev1.RouteIngressCondition) bool {
-	conditionCmpOpts := []cmp.Option{
-		cmpopts.EquateEmpty(),
-		cmpopts.IgnoreFields(routev1.RouteIngressCondition{}, "LastTransitionTime"),
-		cmpopts.SortSlices(func(a, b routev1.RouteIngressCondition) bool { return a.Type < b.Type }),
+	if len(a) != len(b) {
+		return false
+	}
+
+	// Sort both slices by Type to ensure they are in the same order for comparison.
+	sort.Slice(a, func(i, j int) bool { return a[i].Type < a[j].Type })
+	sort.Slice(b, func(i, j int) bool { return b[i].Type < b[j].Type })
```

Comment:
Have you done benchmarks on this code?  [`sort.Slice` uses reflection](https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/sort/slice.go;l=24), albeit the lighter-weight [`reflectlite`](https://pkg.go.dev/internal/reflectlite) rather than [the standard `reflect` package](https://pkg.go.dev/reflect), and even without that, [the underlying quicksort algorithm implementation](https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/sort/zsortfunc.go;l=55-129) is probably slower than a nested loop for the expected number of items (namely 0 or 1 in the usual case, 2 in the exceptional case).  If benchmarks show that `sort.Slice` is fast enough, though, then it doesn't really matter.  (If you do write a benchmark, consider using [`testing.B`](https://pkg.go.dev/testing#hdr-Benchmarks) and adding the benchmark to `pkg/router/controller/status_test.go`.)

Reply:
Good point. I need to back this up with some data. I added a benchmark function (first time I've ever done that) and provided various results in the PR description.

The benchmark with `sort.Slice` results show it's ~500x faster (I'm comparing `ns/op`). I stopped there.

BUT I just tried out nested loops, and I see it's ~5000-7000x faster. You are right, where 2 conditions are our max, nested loops are faster than `sort.Slice`..